src/Context/ContextAwarePluginInterface.php

```diff
@@ -21,10 +21,32 @@
    * already available upon which the definition of subsequent or provided
    * context can be refined.
    *
+   * Note that context gets refined at configuration and execution time of the
+   * plugin.
+   *
    * @param \Drupal\Core\TypedData\DataDefinitionInterface[] $selected_data
    *   An array of data definitions for context that is mapped using a data
    *   selector, keyed by context name.
    */
   public function refineContextDefinitions(array $selected_data);
 
+  /**
+   * Asserts additional metadata for the selected data.
+   *
+   * Allows the plugin to assert additional metadata that is in place when the
+   * plugin has been successfully executed. A typical use-case would be
+   * asserting the node type for a "Node is of type" condition. By doing so,
+   * sub-sequent executed plugins are aware of the metadata and can build upon
+   * it.
+   *
+   * Note that metadata is only asserted on configuration time. The plugin has
+   * to ensure that the run-time data matches the asserted configuration if it
+   * has been executed successfully.
+   *
+   * @param \Drupal\Core\TypedData\DataDefinitionInterface[] $selected_data
+   *   An array of data definitions for context that is mapped using a data
+   *   selector, keyed by context name.
+   */
+  public function assertMetadata(array $selected_data);
```

Comment:
Not super happy about the name "assertMetadata". What does it mean? Actually it should be "refineMetadata" because it changes data types of context? So what is the difference to refineContextDefinitions() then?